internal/lm/fabric.go

```diff
@@ -0,0 +1,156 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package lm
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"math/rand" // nolint:gosec
+	"net"
+	"os"
+	"sort"
+	"strings"
+
+	spec "github.com/NVIDIA/k8s-device-plugin/api/config/v1"
+	"github.com/NVIDIA/k8s-device-plugin/internal/resource"
+
+	"github.com/google/uuid"
+	"k8s.io/klog/v2"
+)
+
+func newImexLabeler(config *spec.Config, devices []resource.Device) (Labeler, error) {
+	if config.Flags.GFD.ImexNodesConfigFile == nil {
+		// No imex config file, return empty labels
+		return empty{}, nil
+	}
+
+	imexConfigFile, err := os.Open(*config.Flags.GFD.ImexNodesConfigFile)
+	if os.IsNotExist(err) {
+		// No imex config file, return empty labels
+		return empty{}, nil
+	} else if err != nil {
+		return nil, fmt.Errorf("failed to open imex config file: %v", err)
+	}
+	defer imexConfigFile.Close()
+
+	clusterUUID, cliqueID, err := getFabricIDs(devices)
+	if err != nil {
+		return nil, err
+	}
+	if clusterUUID == "" || cliqueID == "" {
+		return empty{}, nil
+	}
+
+	imexDomainID, err := getImexDomainID(imexConfigFile)
```

Comment:
In a past iteration we were returning no labels if the file is empty. This doesn't seem to be the case anymore. What is the intended behaviour?

Reply:
If the file is empty, we should not return any labels.

Reply:
added a check